spring-security-oauth2-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/resource/ResourceServerTokenServicesConfiguration.java

```diff
@@ -212,10 +220,12 @@ public UserInfoTokenServices userInfoTokenServices() {
 
 	@Configuration
 	@Conditional(JwkCondition.class)
-	protected static class JwkTokenStoreConfiguration {
+	protected static class JwkTokenStoreConfiguration implements ApplicationContextAware {
```

Comment:
I would remove `implements ApplicationContextAware` in favour of `@Autowired`. This avoids the introduction of the public API `setApplicationContext `.

Reply:
I don't think that is a huge concern since it is a protected class scope.

Reply:
True. I'm not seeing much use of `ApplicationContextAware` and instead `@Autowired`. Is there a preference?

Reply:
I think I would switch to ApplicationContextAware since the aware interfaces are less likely to cause eager instantiation issues (something I have learned since using Autowired).

Reply:
gotchya